src/renderer/features/network/ManageCustomRpcNode/model/add-custom-rpc-model.ts

```diff
@@ -0,0 +1,178 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+import { createForm } from 'effector-forms';
+
+import { networkService } from '@shared/api/network';
+import { ExtendedChain } from '@entities/network';
+import { CheckRpcNodeFxParams, NodeExistParam, RpcCheckResult, SaveRpcNodeFxParams } from '../lib/types';
+import { manageNetworkModel } from '@pages/Settings/Networks/model/manage-network-model';
+import { fieldRules, RpcValidationMapping } from '../lib/constants';
+
+const $addCustomRpcForm = createForm({
+  fields: {
+    name: {
+      init: '',
+      rules: fieldRules.name,
+    },
+    url: {
+      init: '',
+      rules: fieldRules.url,
+    },
+  },
+  validateOn: ['change'],
+});
+
+const formInitiated = createEvent();
+const networkChanged = createEvent<ExtendedChain>();
+const rpcConnectivityChecked = createEvent<RpcCheckResult>();
+const nodeExistChecked = createEvent<boolean>();
+const processStarted = createEvent<boolean>();
+
+const $selectedNetwork = createStore<ExtendedChain | null>(null);
+const $rpcConnectivityResult = createStore<RpcCheckResult>(RpcCheckResult.INIT);
+const $isNodeExist = createStore<boolean>(false);
+const $isProcessStarted = createStore<boolean>(false);
+const $isLoading = createStore<boolean>(false);
+
+const checkRpcNodeFx = createEffect(async ({ network, url }: CheckRpcNodeFxParams): Promise<RpcCheckResult> => {
+  const validationResult = await networkService.validateRpcNode(network.chainId, url);
+  const result = RpcValidationMapping[validationResult];
+
+  return result;
+});
+
+const saveRpcNodeFx = createEffect(async ({ network, form }: SaveRpcNodeFxParams) => {
+  manageNetworkModel.events.rpcNodeAdded({
```

Comment:
These events (and everything related to them) must be moved here
Previously they were moved from the network model, as a first step of becoming a self-sufficient feature
Now time has come to bring them to the final destination 😎
```javascript
rpcNodeAdded
rpcNodeUpdated
rpcNodeRemoved
```

Reply:
Wow you're dragging me in a slippery slope here, expanding the scope of this PR quite a bit... 
I gave it a go, and moved the `rpcNodeAdded` and `rpcNodeUpdated` to see what this would entail. While this seems to be ok, some side effect such as the following are needed and are still in the manage-network-model 

```ts
sample({
  clock: updateConnectionFx.doneData,
  source: networkModel.$providers,
  filter: (_, connection) => {
    return Boolean(connection) && networkUtils.isEnabledConnection(connection!);
  },
  fn: (providers, connection) => {
    const chainId = connection!.chainId;

    return providers[chainId] ? { reconnect: { chainId, providers } } : { start: chainId };
  },
  target: spread({
    start: networkModel.events.chainConnected,
    reconnect: reconnectProviderFx,
  }),
});
```

Anyway, I'll have to push back on this change. This will be for another PR, that I'm happy to take, after this one is merged first. I like to keep PRs as focused as possible and do baby steps to start with.